spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/DefaultCloudSqlJdbcInfoProvider.java

```diff
@@ -36,27 +35,21 @@
 	public DefaultCloudSqlJdbcInfoProvider(GcpCloudSqlProperties properties) {
 		this.properties = properties;
 		Assert.hasText(this.properties.getDatabaseName(), "A database name must be provided.");
-		if (StringUtils.isEmpty(this.properties.getJdbcUrl())) {
```

Comment:
Won't removing this always cause config to error out if instanceConnectionName isn't filled?

Reply:
There is `@Validate` and `@NotEmpty` annotations for the `@ConfigurationProperties` instead: https://docs.spring.io/spring-boot/docs/1.5.7.RELEASE/reference/html/boot-features-external-config.html#boot-features-external-config-validation

Reply:
The issue with using validators here is that `jdbcUrl` can be empty is `instanceConnectionName`, and vice-versa.
I don't think validators have enough expression power to cover that?

Reply:
Well, that can be done via validators as well. See sample here: https://github.com/spring-cloud-stream-app-starters/file/blob/master/spring-cloud-starter-stream-source-file/src/main/java/org/springframework/cloud/stream/app/file/source/FileSourceProperties.java#L90

But I'm fine to leave it as is here.